### PR TITLE
Add analyzers and fix warnings

### PR DESCRIPTION
**Pull request by oddstr13**

### CODE REVIEW

File: Jellyfin.Plugin.KodiSyncQueue/API/KodiSyncQueueController.cs [hunk 1]
```diff
@@ -195,26 +200,29 @@ private List<string> GetAddedOrUpdatedItems(User user, IEnumerable<Guid> ids)
...
+            var itemsAdded = KodiSyncQueuePlugin.Instance.DbRepo.GetItems(dtl, ItemStatus.Added, filters);
+            var itemsRemoved = KodiSyncQueuePlugin.Instance.DbRepo.GetItems(dtl, ItemStatus.Removed, filters);
+            var itemsUpdated = KodiSyncQueuePlugin.Instance.DbRepo.GetItems(dtl, ItemStatus.Updated, filters);
+            var userDataChanged = KodiSyncQueuePlugin.Instance.DbRepo.GetUserInfos(dtl, userId, filters);
+
+            var info = new SyncUpdateInfo(
+                GetAddedOrUpdatedItems(user, itemsAdded),
+                itemsRemoved.Select(id => id.ToString("N", CultureInfo.InvariantCulture)).ToList(),
+                GetAddedOrUpdatedItems(user, itemsUpdated),
+                userDataChanged.Select(i => JsonSerializer.Deserialize<UserItemDataDto>(i.JsonData, JsonDefaults.GetOptions())).ToList());
```
Comment (Bond-009) [inline Jellyfin.Plugin.KodiSyncQueue/API/KodiSyncQueueController.cs:216]: `JsonDefaults.GetOptions()` should be put in a private readonly field (consistent with server and faster)

File: Jellyfin.Plugin.KodiSyncQueue/Jellyfin.Plugin.KodiSyncQueue.csproj [hunk 2]
```diff
@@ -15,7 +17,22 @@
     <PackageReference Include="Jellyfin.Data" Version="10.*-*" />
     <PackageReference Include="Jellyfin.Controller" Version="10.*-*" />
     <PackageReference Include="LiteDB" Version="4.1.4" />
-    <PackageReference Include="Microsoft.AspNetCore.Mvc" Version="2.2.0" />
   </ItemGroup>
 
+  <!-- Code Analyzers-->
+  <ItemGroup>
```
Comment (Bond-009) [inline Jellyfin.Plugin.KodiSyncQueue/Jellyfin.Plugin.KodiSyncQueue.csproj:23]: We generally only include these for debug builds.
Not sure how these behave when publishing a release

**Review by Bond-009 — CHANGES_REQUESTED** [review]

File: Jellyfin.Plugin.KodiSyncQueue/Jellyfin.Plugin.KodiSyncQueue.csproj (hunk 2 above)
Comment (oddstr13) [reply Jellyfin.Plugin.KodiSyncQueue/Jellyfin.Plugin.KodiSyncQueue.csproj:23]: I pulled that part from musicbrainz plugin.
I remember that this was done intentionally at the time, but I don't remember in which repo this was discussed or the reasoning behind it.

I do however generally build release builds, even when testing, and I'm more likely to see the errors from the analyzers this way.

(I'm not saying either way is the right way.)

File: Jellyfin.Plugin.KodiSyncQueue/Data/DbRepo.cs [hunk 3]
```diff
@@ -141,11 +142,10 @@ public void SetUserInfoSync(List<MediaBrowser.Model.Dto.UserItemDataDto> dtos, L
...
             var userInfoCollection = _liteDb.GetCollection<UserInfoRec>(UserInfoCollection);
-            var jsonOptions = JsonDefaults.GetOptions();
 
             dtos.ForEach(dto =>
             {
-                var sJson = System.Text.Json.JsonSerializer.Serialize(dto, jsonOptions);
+                var sJson = System.Text.Json.JsonSerializer.Serialize(dto, _jsonSerializerOptions);
                 _logger.LogDebug("Updating ItemId '{0}' for UserId: '{1}'", dto.ItemId, userId);
 
                 LibItem itemref = itemRefs.FirstOrDefault(x => x.Id.ToString("N", CultureInfo.InvariantCulture) == dto.ItemId);
```
Comment (Bond-009) [inline Jellyfin.Plugin.KodiSyncQueue/Data/DbRepo.cs]: Serialize inside the if statement, use `{@Name}` to serialize to json in the log
Parse the Guid outside the LINQ statement and compare with that

**Review by Bond-009 — CHANGES_REQUESTED** [review]

File: Jellyfin.Plugin.KodiSyncQueue/Data/DbRepo.cs (hunk 3 above)
Comment (oddstr13) [reply Jellyfin.Plugin.KodiSyncQueue/Data/DbRepo.cs]: Not quite sure of what you mean by `{@Name}` here.

This whole plugin is due for a full do-over with a fine-toothed comb for optimizations. I do think that belongs in another PR or three however :wink: 

Getting analyzers going is the first step of me hopefully not screwing up (as bad) when I start poking around some more :see_no_evil:

Comment (Bond-009) [reply Jellyfin.Plugin.KodiSyncQueue/Data/DbRepo.cs]: Nvm, the json string isn't even logged

**Review by Bond-009 — APPROVED** [review]

**Pull request merged**